src/durableorchestrationclient.ts

```diff
@@ -958,7 +974,34 @@ export class DurableOrchestrationClient {
             }
         }
 
-        return this.axiosInstance.get(requestUrl);
+        // If a continuation token is provided, we add it to the request's header
+        let axiosConfig = undefined;
+        if (continuationToken) {
+            axiosConfig = {
+                headers: {
+                    "x-ms-continuation-token": continuationToken,
+                },
+            };
+        }
+
+        // We call the getStatus endpoint and construct a promise callback to handle the recursion

```

Comment:
I _really_ don't like that we have to do this pagination logic. This is really a consequence of us having an indirect proxy to the IDurableClient object via the HTTP apis (even with local RPC). The HTTP APIs are a subset of the C# APIs, meaning we have to do workarounds like this on the worker side, and some of these workarounds (like this one), can have major perf implications.

That being said, I think we have to do this, at least in the short term. I think this points to us needing to have a more comprehensive line using the localRpcEndpoint to IDurableClient implementations (both Durable 1.x and Durable 2.x) before we can get rid of this workaround,  and unfortunately we can't wait for that in the meantime.

Reply:
I've created [this](https://github.com/Azure/azure-functions-durable-extension/issues/1693) ticket to continue this conversation. Thanks for bringing this up :)